binders/kafka-binder/spring-cloud-stream-binder-kafka-reactive/src/test/java/org/springframework/cloud/stream/binder/reactorkafka/ReactorKafkaBinderIntegrationTests.java

```diff
@@ -42,76 +44,86 @@
 import org.springframework.kafka.support.converter.MessagingMessageConverter;
 import org.springframework.kafka.support.converter.RecordMessageConverter;
 import org.springframework.kafka.test.EmbeddedKafkaBroker;
-import org.springframework.kafka.test.condition.EmbeddedKafkaCondition;
 import org.springframework.kafka.test.context.EmbeddedKafka;
 import org.springframework.kafka.test.utils.KafkaTestUtils;
 import org.springframework.messaging.Message;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 /**
+ * Integration tests for {@link ReactorKafkaBinder}.
+ *
  * @author Soby Chacko
  * @author Gary Russell
+ * @author Chris Bono
  */
+@ExtendWith(SpringExtension.class)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
```

Comment:
Went from static single instance to instance per test (there are only 2 tests). The reason is that we now have 2 tests (w/ and w/o KAC excluded) and each of them writes records. Additionally `KafkaTestUtils` does not have an easy way to "get last record" and the 2nd tests always fails w/ "more than 1 record found". To get the last record is possible but the tradeoff on complexity of doing that vs. keeping the test simple and isolated wins. If there were > 2 tests I would consider going the "static kafka" route.